src/toil/jobStores/abstractJobStore.py

```diff
@@ -903,6 +904,15 @@ def writeFileStream(self, jobStoreID=None, cleanup=False, basename=None):
                file basename so that when searching the job store with a query
                matching that basename, the file will be detected.
 
+        :param str mode: an optional string that specifies the mode in which the file is opened.
+                It defaults to 'b' which means open for reading in binary mode.
```

Comment:
This needs to define the supported mode characters. Do I pass `'text'` or `'t'` for text mode? Or just an empty string?